llvm/lib/Target/AIE/AIECombinerHelper.cpp

```diff
@@ -901,6 +916,31 @@ bool modifyToS20(InstrNode Start, MachineRegisterInfo &MRI, MachineIRBuilder &B,
     Helper.tryCombineCopy(*StartNodeMI);
     return true;
   }
+  case TargetOpcode::G_INTRINSIC: {
+    if (isS20Intrinsic(*StartNodeMI, MRI)) {
+      Register ExtractDstReg = StartNodeMI->getOperand(0).getReg();
+      // Note: Operand 1 is the ID of the intrinsic
+      const Register SrcReg0 = StartNodeMI->getOperand(2).getReg();
+      const Register SrcReg1 = StartNodeMI->getOperand(3).getReg();
+      const Register SignReg = StartNodeMI->getOperand(4).getReg();
```

Comment:
We should not do like this; this is very error prone. 
We might add a new intrinsic in inS20Intrinsic and this would then fail  if we do not have the required number of operands.